backend/db/utils.py

```diff
@@ -137,7 +137,7 @@ def refresh_derived_tables(
      entry will appear further down in the list."""
     temp_table_suffix = '_new'
     ddl_modules_queue = derived_tables_queue
-    views = [x for x in VIEWS_TO_REFRESH if x in ddl_modules_queue]
+    views = [x for x in list_views() if x in ddl_modules_queue]
```

Comment:
## The bugfix
`VIEWS_TO_REFRESH` was an empty list because we didn't care to refresh. But this created a problem where, because the way things are coded, this variable was not being read at the time of creating derived tables/views, but only when deleting them. So this resulted in the new view being created and the old one backed up, but during cleanup where old tables/views were removed, this variable was empty, so the view was not removed, and then the table it depends on could not be dropped because of lack of cascade.